sec2_the-ruby-programming-language/2.12_final_ruby_project/2_bcrypt_autheticate.rb

```diff
@@ -0,0 +1,45 @@
+require 'bcrypt'
+
+users = [
+  {username: "michelle", password: "password1"},
+  {username: "apple", password: "password2"},
+  {username: "data", password: "password3"},
+  {username: "delta", password: "password4"},
+  {username: "alpha", password: "password5"},
+  ]
+
+# Hash the password
+def create_hash_digest(password)
+  BCrypt::Password.create(password)
+end
+
+# Verify the password
+def verify_hash_digest(password)
+  BCrypt::Password.new(password)
+end
```

Comment:
What is the difference between `.create` and `.new` in Line 13, Line 18

Reply:
These are 2 methods in the bcrypt library.
`new` is `initialize` (unless they rewrite it). But create can be anything.
Please go to their docs / code and find out what each does and comment here :)

Reply:
`.new`/ `.initialise` will check if the validity of the raw hash created, split it using `$` and return the separated hash while `.create` will calculate the cost factor of the hash and create a hash and pass it to `.initialise`.